### DISCUSSION TITLE
State Management

### DISCUSSION THREAD

**Discussion by TheHollidayInn** [hoppscotch/hoppscotch#2793] [Q&A] [answered]
Is there a reason the project doesn't use Pinia or Veux? The collections passes props and events which would be probably simplified by the two.

**Comment by AndrewBastin** [comment, marked as answer]
Well, we had legacy reasons regarding our code structure with a lot of code being out of components needing to require reactivity and Vue 2 not having a clean system for it at the time.

With the advent of Vue 3 and its `@vue/reactivity` system, we are looking into revamping the store system to use Pinia moving on. But it is a fairly lower priority at the moment.
